sympy/functions/elementary/hyperbolic.py

```diff
@@ -287,19 +287,19 @@ def _eval_is_positive(self):
         re, im = arg.as_real_imag()
         im_mod = im % (2*pi)
 
-        if im_mod == 0:
+        if im_mod.is_zero:
             return True
 
-        if re == 0:
+        if re.is_zero:
             if im_mod < pi/2 or im_mod > 3*pi/2:
                 return True
             elif im_mod >= pi/2 or im_mod <= 3*pi/2:
                 return False
```

Comment:
The problem here is with using inequalities directly in conditionals. This will raise if the inequalities are indeterminate (see the example from the previous PR).